src/app/application/components/data-source/DataLoader.vue

```diff
@@ -71,14 +71,13 @@ import ErrorBlock from '@/app/common/ErrorBlock.vue'
 import LoadingBlock from '@/app/common/LoadingBlock.vue'
 
 const props = withDefaults(defineProps<{
-  data: any[]
-  errors: (Error | undefined)[]
+  data?: any[]
+  errors?: (Error | undefined)[]
   src: string
   loader?: boolean
 }>(), {
   errors: () => [],
   data: () => [],
-  src: '',
```

Comment:
How come you removed this?

Reply:
`props.src` is not optional and this will never be useful. Not providing `props.src` is a user error. Default values for props are for optional props.

Reply:
`props.src` is optional, you purposefully don't have to add a `:src` to a DataLoader, you can just pass it data from other DataSources if you like.

If it would have to have a sync on the thinking behind the interactions/relationships with DataSource/DataLoader (or at least the intended functionality 😅 ) let me know.

Reply:
So then why is `props.src` not optional in the definition?

Reply:
That would be the better fix here, looks like I somehow missed that when I initially added this. I guess the examples I added happened to all have `:src`s

Reply:
I changed this now.

Reply:
Awesome, thanks!